### ISSUE TITLE
Info event not firing

### ISSUE THREAD

**Issue by anajmudeen** [fent/node-ytdl-core#382]
https://github.com/fent/node-ytdl-core/blob/e2cb8113df0eeb751f358ce895fa596db353a952/lib/index.js#L70

**Comment by anajmudeen** [comment]
Below code is working as expected. As like error event in https://github.com/fent/node-ytdl-core/blob/e2cb8113df0eeb751f358ce895fa596db353a952/lib/index.js#L65

`setImmediate(() => {
    stream.emit('info', info, format);
  });`

**Comment by fent** [comment]
Are you saying it's not firing the `info` event if it can't find a format with the given options?

**Comment by anajmudeen** [comment]
No, Its firing the info event with formats. But its not received by the videoObject.on('info',eventHandler);

**Comment by anajmudeen** [comment]
Below code is not receiving 'info' event.
`videoObject.on('info', (info, format) => {
                console.log("On info:", info, format);
            });`

**Comment by fent** [comment]
`videoObject` is the stream returned from ytdl? Can I see more of your code on how you're calling ytdl?

**Comment by fent** [comment]
Still not able to reproduce. The exact same code fires the `info` event. Are you sure you're on latest?

**Comment by anajmudeen** [comment]
Yes, am on latest. Will it work if the above code inside the async function. Like below code.
```
function getVideo(videoId){
return new Promise(async (resolve,reject)=>{
let videoObject = ytdl(videoId);
        videoObject
            .on('progress',(chunkLength, downloaded, total) => {
                let percent = Math.floor((downloaded / total) * 100);
                console.log("Progress:",percent);
            });

        videoObject.on('info', (info, format) => {
            console.log("On info:", info, format);
        });

        videoObject.on('finish', () => {
           //Some await call will be here, thats why i use async in promise callback
            resolve();
        });
});
}
```

**Comment by fent** [comment]
I don't see why it wouldn't work putting it anywhere. Does `progress` fire? btw, I don't think you can make the promise callback async, it's not awaited when called.

**Comment by anajmudeen** [comment]
Progress event working as expected.

**Comment by fent** [comment]
Any update?

> Below code is working as expected. As like error event in
> 
> [node-ytdl-core/lib/index.js](https://github.com/fent/node-ytdl-core/blob/e2cb8113df0eeb751f358ce895fa596db353a952/lib/index.js#L65)
> 
> Line 65 in [e2cb811](/fent/node-ytdl-core/commit/e2cb8113df0eeb751f358ce895fa596db353a952)
> 
>  setImmediate(() => { 
> `setImmediate(() => { stream.emit('info', info, format); });`

What does this mean? Does replacing the code with this have it work?
